lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -0,0 +1,909 @@
+//===- HandshakeToFIRRTL.cpp - Translate Handshake into FIRRTL ------------===//
+//
+// Copyright 2019 The CIRCT Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+#include "circt/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.h"
+#include "circt/Dialect/FIRRTL/Ops.h"
+#include "circt/Dialect/Handshake/HandshakeOps.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/Function.h"
+#include "mlir/IR/OpImplementation.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/IR/Types.h"
+#include "mlir/IR/Value.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "mlir/Transforms/Passes.h"
+#include "mlir/Transforms/Utils.h"
+
+using namespace mlir;
+using namespace mlir::handshake;
+using namespace circt;
+using namespace circt::firrtl;
+
+using ValueVector = std::vector<Value>;
+using ValueVectorList = std::vector<ValueVector>;
```

Comment:
Not important for this patch to land but FYI `vector<vector<pointer>>` is very inefficient.  You might consider something like `vector<SmallVector<pointer, 4>>`.